cmd/veneur-prometheus/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math"
+	"net/http"
+	"time"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/Sirupsen/logrus"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+)
+
+var (
+	debug       = flag.Bool("d", false, "Enable debug mode")
+	metricsHost = flag.String("h", "http://localhost:9090/metrics", "The full URL — like 'http://localhost:9090/metrics' to query for Prometheus metrics.")
+	interval    = flag.String("i", "10s", "The interval at which to query. Value must be parseable by time.ParseDuration (https://golang.org/pkg/time/#ParseDuration).")
+	statsHost   = flag.String("s", "127.0.0.1:8126", "The host and port — like '127.0.0.1:8126' — to send our metrics to.")
+)
+
+func main() {
+	flag.Parse()
+
+	c, _ := statsd.New(*statsHost)
+
+	if *debug {
+		logrus.SetLevel(logrus.DebugLevel)
+	}
+	logrus.Debug("HELLO")
+
+	i, err := time.ParseDuration(*interval)
+	if err != nil {
+		logrus.WithError(err).Fatalf("Failed to parse interval '%s'", *interval)
+	}
+
+	ticker := time.NewTicker(i)
+	for _ = range ticker.C {
+		collect(c)
+	}
+}
+
+func collect(c *statsd.Client) {
+	logrus.WithFields(logrus.Fields{
+		"stats_host":   *statsHost,
+		"metrics_host": *metricsHost,
+	}).Debug("Beginning collection")
+
+	resp, _ := http.Get(*metricsHost)
+	d := expfmt.NewDecoder(resp.Body, expfmt.FmtText)
+	var mf dto.MetricFamily
+	for {
+		err := d.Decode(&mf)
```

Comment:
I'm assuming this decoder is guaranteed to be exact? Otherwise we'll get a lot of problems....